zerver/lib/actions.py

```diff
@@ -4435,16 +4435,26 @@ def check_invite_limit(user: UserProfile, num_invitees: int) -> None:
 
         default_max = settings.INVITES_DEFAULT_REALM_DAILY_MAX
         newrealm_age = datetime.timedelta(days=settings.INVITES_NEW_REALM_DAYS)
-        if (user.realm.date_created > timezone_now() - newrealm_age
-                and user.realm.max_invites <= default_max):
-            new_realms = Realm.objects.filter(
-                date_created__gte=timezone_now() - newrealm_age,
-                _max_invites__lte=default_max,
-            ).all()
-            for days, count in settings.INVITES_NEW_REALM_LIMIT_DAYS:
-                recent_invites = estimate_recent_invites(new_realms, days=days)
-                if num_invitees + recent_invites > count:
-                    raise InvitationError(msg, [], sent_invitations=False)
+        if user.realm.date_created - timezone_now() > newrealm_age:
+            # If the user's realm is sufficiently old, then we don't
+            # apply invite limits.  This code is targeted at the specific
```

Comment:
We do still apply invite limits -- see the `raise` above. Which is important. :-)

Ditto on the "invite limits" in the next comment.

I think potentially a way to help make this logic clearer would be to pull this whole stanza (from `INVITES_DEFAULT_REALM_DAILY_MAX` to the end of the function) into a separate function, with a name like maybe `check_aggregate_invite_limit`. Then also these two comments can say "this limit" or "the aggregate invite limit".

Reply:
Also possibly helpful for clarity: that function doesn't need `user` as a parameter, only `realm=user.realm`.

Actually, I guess the same is true of this `check_invite_limit` as a whole -- it's all about applying limits at the realm level and up.

Reply:
Another idea for clarifying this: make the `if settings.OPEN_REALM_CREATION` at the top an early-return too. This whole function is for when the server allows anyone on the network to create a realm, because it's for spam where the realm itself belongs to the spammer.